File: 136. Unique Number/singleNumber.py
'''
Leet Code problem 136. Unique Number
https://leetcode.com/problems/single-number/
'''
from collections import Counter
from typing import List


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        counter = Counter(nums)
        uniques = [item for item, count in counter.items() if count == 1]
        return uniques[0]

File: 136. Unique Number/cs/SingleNumber/SingleNumber.cs
/*
Leet Code problem 136. Unique Number
https://leetcode.com/problems/single-number/
*/

namespace LeetCode.SingleNumber;

public class Solution
{
    public int SingleNumber(int[] nums)
    {
        int result = 0;
        foreach (var n in nums)
            result ^= n;
        return result;
    }
}
